src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -282,13 +237,22 @@ def new_posix_test_path(self, shared=False, create=True, parent=None, mount_dir_
             if not run_remote(self.log, self.hostlist_clients, cmd).passed:
                 self.fail(f"Failed to mkdir {path}")
 
+        # Remove the path during test tearDown. Not needed when using the path as a mount point as
+        # cleanup_path() is called as part of cleanup_mounted_path()
```

Comment:
nit - I don't think we need a "meta" comment here. I would expect `cleanup_mounted_path` to be self-contained. I.e. I wouldn't think to call BOTH `cleanup_path` and `cleanup_mounted_path` on the same path